packages/cms-cli/commands/functions/list.js

```diff
@@ -0,0 +1,81 @@
+const { getRoutes } = require('@hubspot/cms-lib/api/function');
+const { logger } = require('@hubspot/cms-lib/logger');
+const {
+  logServerlessFunctionApiErrorInstance,
+  ApiErrorContext,
+} = require('@hubspot/cms-lib/errorHandlers');
+const { outputFunctions } = require('@hubspot/cms-lib/lib/functions');
+const {
+  loadConfig,
+  validateConfig,
+  checkAndWarnGitInclusion,
+} = require('@hubspot/cms-lib');
+
+const {
+  addConfigOptions,
+  addPortalOptions,
+  addUseEnvironmentOptions,
+  getPortalId,
+  setLogLevel,
+} = require('../../lib/commonOpts');
+const { trackCommandUsage } = require('../../lib/usageTracking');
+const { logDebugInfo } = require('../../lib/debugInfo');
+const { validatePortal } = require('../../lib/validation');
+
+const loadAndValidateOptions = async options => {
+  setLogLevel(options);
+  logDebugInfo(options);
+  const { config: configPath } = options;
+  loadConfig(configPath, options);
+  checkAndWarnGitInclusion();
+
+  if (!(validateConfig() && (await validatePortal(options)))) {
+    process.exit(1);
+  }
+};
+
+exports.command = 'list';
+exports.describe = 'List currently deployed functions';
+
+exports.handler = async options => {
+  loadAndValidateOptions(options);
+
+  const { json, compact } = options;
+  const portalId = getPortalId(options);
+
+  trackCommandUsage('functions-list', { json, compact }, portalId);
+
+  logger.debug('Getting currently deployed functions');
+
+  const routesResp = await getRoutes(portalId).catch(async e => {
+    await logServerlessFunctionApiErrorInstance(
+      portalId,
+      e,
+      new ApiErrorContext({ portalId })
+    );
+    process.exit();
+  });
+
+  logger.debug(`Retrieving logs for functionId: ${routesResp.id}`);
+
+  return outputFunctions(routesResp, options);
+};
+
+exports.builder = yargs => {
+  addConfigOptions(yargs, true);
```

Comment:
Not sure if it would work well or not, but it'd be sweet to be able to filter results like you can in bash.

E.g. if you type `ls he` and press `tab`, if a folder matches `/.*(he)+.*/gi` it would show that.  So a folder named `help` would display, but a folder named `foo` would not.

Could even be cool to have the `tab` command trigger the API call (and maybe cache results to avoid excessive calls), on which you can locally filter based on the user input from the positional

Reply:
The idea being to make this functionality work similarly to native terminal `ls`